test/e2e/registration_service_test.go

```diff
@@ -390,6 +390,40 @@ func (s *registrationServiceTestSuite) TestSignupOK() {
 	})
 }
 
+func (s *registrationServiceTestSuite) TestUserSignupFoundWhenNamedWithEncodedUsername() {
+	hostAwait := s.Host()
+
+	// Create a token and identity to sign up with, but override the sub claim with "arnold" so that we create a UserSignup
+	// with that name
+	identity0 := authsupport.NewIdentity()
+	emailValue := "arnold@acme.com"
+	emailClaim0 := authsupport.WithEmailClaim(emailValue)
+	token0, err := authsupport.GenerateSignedE2ETestToken(*identity0, emailClaim0, authsupport.WithSubClaim("arnold"))
+	require.NoError(s.T(), err)
+
+	// Call the signup endpoint
+	invokeEndpoint(s.T(), "POST", s.route+"/api/v1/signup", token0, "", http.StatusAccepted)
+
+	// Wait for the UserSignup to be created
+	userSignup, err := hostAwait.WaitForUserSignup("arnold",
+		wait.UntilUserSignupHasConditions(ConditionSet(Default(), PendingApproval())...),
+		wait.UntilUserSignupHasStateLabel(toolchainv1alpha1.UserSignupStateLabelValuePending))
```

Comment:
it's in pending approval because the automatic approval is disabled for e2e tests by default, is that correct?

Reply:
Right.